File: 2017/src/main/java/frc/robot/subsystems/Lights.java
package frc.robot.subsystems;

import com.ctre.phoenix.CANifier;
import com.ctre.phoenix.CANifier.LEDChannel;

import edu.wpi.first.wpilibj.command.Subsystem;
import frc.robot.RobotMap;
import frc.robot.commands.LightRainbow;

public class Lights extends Subsystem {
	private CANifier m_canifier = new CANifier(RobotMap.CAN.CANIFIER);
	
	public Lights() {
	}

	@Override
	public void initDefaultCommand() {
		setDefaultCommand(new LightRainbow());
	}

	public void setColour(double r, double g, double b) {
		m_canifier.setLEDOutput(RobotMap.Robot.LED_BRIGHTNESS * r, LEDChannel.LEDChannelA);
		m_canifier.setLEDOutput(RobotMap.Robot.LED_BRIGHTNESS * g, LEDChannel.LEDChannelB);
		m_canifier.setLEDOutput(RobotMap.Robot.LED_BRIGHTNESS * b, LEDChannel.LEDChannelC);
	}
}

File: 2019_cpp/src/main/cpp/subsystems/Lights.cpp
#include "subsystems/Lights.h"

#include "RobotMap.h"
#include "commands/LightRainbow.h"

Lights::Lights() :
	frc::Subsystem("Lights"),
	m_canifier(std::make_unique<CANifier>(0))
{
}

void Lights::InitDefaultCommand()
{
	SetDefaultCommand(new LightRainbow());
}

void Lights::SetColour(const double &r, const double &g, const double &b)
{
	m_canifier->SetLEDOutput(kLedBrightness * r, CANifier::LEDChannel::LEDChannelA);
	m_canifier->SetLEDOutput(kLedBrightness * g, CANifier::LEDChannel::LEDChannelB);
	m_canifier->SetLEDOutput(kLedBrightness * b, CANifier::LEDChannel::LEDChannelC);
}
